### DISCUSSION TITLE
Streamlink with OpenPLI 9

### DISCUSSION THREAD

**Discussion by ChefSpeciale** [streamlink/streamlink#5574] [Q&A] [answered]
### Description

Hi all,

I'm trying to build streamlink in openpli 9, i use python 3.9.9 with pip.

I'm have some problems, its keeps saying that i have not installed libxml2 and libxslt development packages..

```
Using cached lxml-4.9.3.tar.gz (3.6 MB)
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/var/volatile/tmp/pip-install-y6hter6c/lxml/setup.py'"'"'; __file__='"'"'/var/volatile/tmp/pip-install-y6hter6c/lxml/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /var/volatile/tmp/pip-install-y6hter6c/lxml/pip-egg-info
         cwd: /var/volatile/tmp/pip-install-y6hter6c/lxml/
    Complete output (5 lines):
    /var/volatile/tmp/pip-install-y6hter6c/lxml/setup.py:67: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
      import pkg_resources
    Building lxml version 4.9.3.
    Building without Cython.
    Error: Please make sure the libxml2 and libxslt development packages are installed.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```
lscpu:
```
Architecture:        armv7l
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  1
Core(s) per socket:  2
Socket(s):           1
Vendor ID:           Broadcom
Model:               3
Model name:          Brahma B15
Stepping:            0x0
BogoMIPS:            54.00
Flags:               half thumb fastmult vfp edsp thumbee neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm
```

Maybe someone can help me:) Thx

**Comment by bastimeyer** [comment, marked as answer]
`lxml` is one of Streamlink's dependencies. You're having issues installing `lxml`, not Streamlink.
https://lxml.de/

lxml doesn't provide wheels (pre-built binary Python packages) for `armv7l`. It only has wheels for `x86_64`, `i686` and `aarch64`, for glibc 2.17 / 2.28 and musl libc 1.1, as you can see here:
https://pypi.org/project/lxml/4.9.3/#files

This means you'll need to build lxml from source, which is what `pip` tried to do in your case, as it couldn't find matching wheels. `libxml2` and `libxslt` are two of `lxml`'s build-dependencies, as well as `Cython` (optional). Make sure that you have the development packages of those two libraries installed via your distro's package management. Otherwise, you'll have to build those libraries yourself first.

I don't know OpenPLi, so you'll have to ask there or figure it out on your own.
